src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -95,23 +94,39 @@ private void Commit(PresentationItem item, Model model, char? commitChar, Cancel
                         IgnoreTrimWhiteSpace = EditOptions.DefaultMinimalChange.DifferenceOptions.IgnoreTrimWhiteSpace
                     });
 
-                    // edit subject buffer (not view) because text changes are in terms of current document.
-                    using (var textEdit = this.SubjectBuffer.CreateEdit(editOptions, reiteratedVersionNumber: null, editTag: null))
+                    var textChangesInCurrentViewSnapshot = new List<TextChange>();
+
+                    foreach (var triggerSnapshotTextChange in textChangesAgainstTriggerSnapshot)
                     {
-                        for (int iChange = 0; iChange < textChanges.Length; iChange++)
+                        var currentSpanInBuffer = new SnapshotSpan(
+                            model.TriggerSnapshot,
+                            triggerSnapshotTextChange.Span.ToSpan());
+
+                        // Try mapping the item span against the trigger snapshot to the ViewBuffer's trigger snapshot.
+                        // Then map that forward to the ViewBuffer's current snapshot.
+                        var originalSpanInView = model.GetViewBufferSpan(triggerSnapshotTextChange.Span).TextSpan.ToSnapshotSpan(model.TriggerSnapshot);
```

Comment:
This line looks wrong to me.  We have a span relative to the snapshot for the subject buffer .  We then map that to the view buffer's snapshot.    BUt then we just ask for that span against the trigger snapshot... 

The trigger snapshot corresponds to the subject buffer.  But the view buffer is totallly different...